MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/MoleculeAndAxisWidget.py

```diff
@@ -0,0 +1,112 @@
+
+    If one index is given, the molecule axis will be a vector from
+    the molecule centre to the atom with the given index.
+
+    If two indices are given, the molecule axis will be a vector
+    between the atoms with the two indices.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Populate the combo box with valid molecule names."""
+        super().__init__(*args, index_boxes=2, **kwargs)
+        self.enable_combo_boxes()
+        for cbox in self.index_combo_boxes:
+            cbox.currentIndexChanged.connect(self.enable_combo_boxes)
+        self._field.currentTextChanged.connect(self.update_combo_boxes)
+
+    @Slot()
+    def molecule_changed(self):
+        """Update atom index combo boxes."""
+        super().molecule_changed()
+        self.update_combo_boxes()
+
+    @Slot()
+    def update_combo_boxes(self):
+        """Set new atom index limits on molecule change."""
+        if self.selected_mol is None:
+            for cbox in self.index_combo_boxes:
+                cbox.clear()
+                cbox.addItems(["None"])
+                cbox.setCurrentText("None")
+                cbox.setEnabled(False)
+            return
+        n_atoms = len(self.selected_mol["atom_indices"])
+        index_elements = ["None"]
+        index_elements.extend(str(x) for x in range(n_atoms))
+        for box_index, cbox in enumerate(self.index_combo_boxes):
+            cbox: QComboBox
+            current_value = cbox.currentText()
+            cbox.clear()
+            cbox.addItems(index_elements)
+            cbox.setCurrentText("None") if box_index else cbox.setCurrentText("0")
+            try:
+                numval = int(current_value)
+            except (TypeError, ValueError):
+                continue
+            else:
+                if numval in range(n_atoms):
+                    cbox.setCurrentText(current_value)
+
+    def enable_combo_boxes(self):
+        """Activate or deactivate combo boxes based on other inputs.
+
+        The combo boxes of atom index are disabled if no molecule has been
+        selected. If the first index is already set to 'None', the second
+        combo box is disabled, since its index value will not be used.
+        """
+        if self.selected_mol is None:
+            for cbox in self.index_combo_boxes:
+                cbox.setEnabled(False)
+            return
+        for cbox in self.index_combo_boxes:
+            cbox.setEnabled(True)
+        if self.index_combo_boxes[0].currentText() == "None":
+            self.index_combo_boxes[1].setEnabled(False)
+
+    def get_widget_value(self) -> list[str, int | None, int | None]:
+        """Get the molecule name and atom indices.
+
+        Returns
+        -------
+        list[str, int | None, int | None]
+            Molecule name, optional atom index 1, optional atom index 2
+
+        """
+        result = [super().get_widget_value(), None, None]
+        for nbox, cbox in enumerate(self.index_combo_boxes):
```

Comment:
Even if you don't choose the solution above, `enumerate` takes an optional startindex.
```suggestion
        for nbox, cbox in enumerate(self.index_combo_boxes, 1):
```
Which means you can just use
```python
result[nbox] = index
```
below